java/bogdana_maksiuta/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -0,0 +1,59 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class BinarySearchTree {
+    private Node rootNode;
+
+    static class Node {
+        private Node leftChildNode;
+        private Node rightChildNode;
+        private int value;
+
+        public Node getLeftChildNode() {
+            return leftChildNode;
+        }
+
+        public void setLeftChildNode(Node leftChildNode) {
+            this.leftChildNode = leftChildNode;
+        }
+
+        public Node getRightChildNode() {
+            return rightChildNode;
+        }
+
+        public void setRightChildNode(Node rightChildNode) {
+            this.rightChildNode = rightChildNode;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public void setValue(int value) {
+            this.value = value;
+        }
+    }
+
+
+    public void addNode(int value) {
+        Node node = new Node();
+        node.setValue(value);
+        if (rootNode == null) {
+            rootNode = node;
+        } else {
+            Node currentNode = rootNode;
+            Node parentNode;
+            parentNode = currentNode;
+            if (value < currentNode.getValue()) {
+                currentNode = currentNode.getLeftChildNode();
+                if (currentNode == null) {
```

Comment:
it seems we won't insert the value if a currentNode is not null; I think the code should also contain a loop or recursion that will allow us to insert values into lower levels of a tree